monitoring/defender/autotask/on_module_factory_events.js

```diff
@@ -0,0 +1,1014 @@
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "address",
+        name: "_guard",
+        type: "address",
+      },
+    ],
+    name: "setGuard",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "uint256",
+        name: "bond",
+        type: "uint256",
+      },
+    ],
+    name: "setMinimumBond",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "uint32",
+        name: "cooldown",
+        type: "uint32",
+      },
+    ],
+    name: "setQuestionCooldown",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "uint32",
+        name: "timeout",
+        type: "uint32",
+      },
+    ],
+    name: "setQuestionTimeout",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "address",
+        name: "_target",
+        type: "address",
+      },
+    ],
+    name: "setTarget",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "uint256",
+        name: "templateId",
+        type: "uint256",
+      },
+    ],
+    name: "setTemplate",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "bytes",
+        name: "initParams",
+        type: "bytes",
+      },
+    ],
+    name: "setUp",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [],
+    name: "target",
+    outputs: [
+      {
+        internalType: "address",
+        name: "",
+        type: "address",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    inputs: [],
+    name: "template",
+    outputs: [
+      {
+        internalType: "uint256",
+        name: "",
+        type: "uint256",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "address",
+        name: "newOwner",
+        type: "address",
+      },
+    ],
+    name: "transferOwnership",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+exports.handler = async function (event) {
+  console.log(
+    "\n\n/************************** EVENT *****************************/"
+  );
+  console.log(JSON.stringify(event));
+  console.log(
+    "/**********************************************************/\n\n"
+  );
+  if (event && event.request && event.request.body) {
+    // variables from autotask creation
+    const nodeProvider = "{{nodeProvider}}";
+    const discordWebHookUrl = "{{discordWebHookUrl}}";
+    const transaction = event.request.body.transaction;
```

Comment:
Instead of loading the transaction and iterating over its logs, we should get the information from the event (input parameter) or use that information to be able to pinpoint the exact event. Instead of getting all logs from the tx and iterating. There could be more than one ModuleProxyCreation event.